evalml/tests/data_checks_tests/test_natural_language_nan_data_check.py

```diff
@@ -0,0 +1,112 @@
+import numpy as np
+import pandas as pd
+import woodwork as ww
+
+from evalml.data_checks import (
+    DataCheckError,
+    DataCheckMessageCode,
+    NaturalLanguageNaNDataCheck
+)
+
```

Comment:
General comment, we check using None here. While it shouldn't make a difference in behavior, can we also check with `np.nan`?

Reply:
Can't use `np.nan` here since their will be conflicting types but I will go ahead and try with empty string.